protocols/gossipsub/tests/smoke.rs

```diff
@@ -187,7 +187,7 @@ fn multi_hop_propagation() {
     let _ = env_logger::try_init();
 
     fn prop(num_nodes: u8, seed: u64) -> TestResult {
-        if num_nodes < 2 || num_nodes > 50 {
+        if !(2..=50).contains(&num_nodes) {
```

Comment:
Not that it is important, though I am curious, do folks find this easier to read?

Reply:
Pretty sure this one was done by clippy, I don't remember writing that.

If I were to write this test, I would probably split it into two `if`s:

```rust
if num_nodes < 2 {
	return TestResult::discard();
}

if num_nodes > 50 {
	return TestResult::discard();
}
```

Reply:
Or make a newtype that implements `Arbitrary` that only generates numbers between 3 and 49. We can call it `NonZeroOneTwoU6Minus15`

Reply:
:+1: on the above suggestions in general, though I think in this particular case leaving it as is is just fine.